src/components/buttons/button-secondary.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { buttonStyles } from './buttons-styles';
+
+interface SecondaryButtonProps {
+  children: React.ReactNode;
+  icon?: React.ReactNode;
+  hasIcon: boolean;
+  size: 'default' | 'compact';
```

Comment:
como vc só tem 2 sizes, vc pode fazer uma prop boolean `compact`, aí não precisa ficar comparando string
Além disso, em casos em que for comparar strings, é interessante criar um enum